### ISSUE TITLE
Bug: Invoking a lambda locally makes Docker break

### ISSUE THREAD

**Issue by andrybicio** [aws/aws-sam-cli#5014]
### Description:
Whenever I start an API gateway with some lambdas defined in a SAM template.yml that is valid and deployed correctly to the cloud, I cannot trigger the endpoints anymore since it returns errors about Docker connections. 

### Steps to reproduce:
simply run 
```
sam local start-api -n local.json -p 3001
```

### Observed result:

This is the output with debug flag activated
```
2023-04-14 12:33:01,109 | Config file location: /home/andrea/authentication-template/backend/samconfig.toml
2023-04-14 12:33:01,109 | Config file '/home/andrea/authentication-template/backend/samconfig.toml' does not exist
2023-04-14 12:33:01,146 | Using SAM Template at /home/andrea/authentication-template/backend/template.yml
2023-04-14 12:33:01,206 | Using config file: samconfig.toml, config environment: default
2023-04-14 12:33:01,207 | Expand command line arguments to:
2023-04-14 12:33:01,207 | --template_file=/home/andrea/authentication-template/backend/template.yml --env_vars=local.json --port=3001 --host=127.0.0.1 --static_dir=public --layer_cache_basedir=/home/andrea/.aws-sam/layers-pkg --container_host=localhost --container_host_interface=127.0.0.1 
2023-04-14 12:33:01,337 | local start-api command is called
2023-04-14 12:33:01,378 | No Parameters detected in the template
2023-04-14 12:33:01,412 | There is no customer defined id or cdk path defined for resource AuthStackBucket, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,412 | There is no customer defined id or cdk path defined for resource Api, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource HelloWorld, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource PrintRequestContext, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource KMSKey, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource UserPool, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource SESEmailIdentity, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource UserPoolClient, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource PreSignup, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource CustomEmailSender, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource OnCognitoSignupPermission, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource AuthTemplateTable, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,413 | There is no customer defined id or cdk path defined for resource CreateOrganization, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,417 | 0 stacks found in the template
2023-04-14 12:33:01,417 | No Parameters detected in the template
2023-04-14 12:33:01,444 | There is no customer defined id or cdk path defined for resource AuthStackBucket, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,444 | There is no customer defined id or cdk path defined for resource Api, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource HelloWorld, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource PrintRequestContext, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource KMSKey, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource UserPool, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource SESEmailIdentity, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource UserPoolClient, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource PreSignup, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource CustomEmailSender, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource OnCognitoSignupPermission, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,445 | There is no customer defined id or cdk path defined for resource AuthTemplateTable, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,446 | There is no customer defined id or cdk path defined for resource CreateOrganization, so we will use the resource logical id as the resource id
2023-04-14 12:33:01,449 | 13 resources found in the stack 
2023-04-14 12:33:01,449 | Found Serverless function with name='HelloWorld' and CodeUri='.'
2023-04-14 12:33:01,449 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 12:33:01,449 | Found Serverless function with name='PrintRequestContext' and CodeUri='.'
2023-04-14 12:33:01,450 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 12:33:01,450 | Found Serverless function with name='PreSignup' and CodeUri='.'
2023-04-14 12:33:01,450 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 12:33:01,450 | Found Serverless function with name='CustomEmailSender' and CodeUri='.'
2023-04-14 12:33:01,450 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 12:33:01,450 | Found Serverless function with name='CreateOrganization' and CodeUri='.'
2023-04-14 12:33:01,450 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 12:33:01,461 | Detected Inline Swagger definition
2023-04-14 12:33:01,461 | Lambda function integration not found in Swagger document at path='/info' method='get'
2023-04-14 12:33:01,461 | Lambda function integration not found in Swagger document at path='/organizations' method='post'
2023-04-14 12:33:01,461 | Lambda function integration not found in Swagger document at path='/hello' method='get'
2023-04-14 12:33:01,461 | Found '0' APIs in resource 'Api'
2023-04-14 12:33:01,461 | Found '1' API Events in Serverless function with name 'HelloWorld'
2023-04-14 12:33:01,461 | Found '1' API Events in Serverless function with name 'PrintRequestContext'
2023-04-14 12:33:01,461 | Found '0' API Events in Serverless function with name 'PreSignup'
2023-04-14 12:33:01,462 | Found '0' API Events in Serverless function with name 'CustomEmailSender'
2023-04-14 12:33:01,462 | Found '1' API Events in Serverless function with name 'CreateOrganization'
2023-04-14 12:33:01,462 | Removed duplicates from '3' Explicit APIs and '0' Implicit APIs to produce '3' APIs
2023-04-14 12:33:01,462 | 3 APIs found in the template
2023-04-14 12:33:01,475 | Mounting PrintRequestContext at http://127.0.0.1:3001/info [GET, OPTIONS]
2023-04-14 12:33:01,475 | Mounting HelloWorld at http://127.0.0.1:3001/hello [GET, OPTIONS]
2023-04-14 12:33:01,475 | Mounting CreateOrganization at http://127.0.0.1:3001/organizations [POST, OPTIONS]
2023-04-14 12:33:01,475 | You can now browse to the above endpoints to invoke your functions. You do not need to restart/reload SAM CLI while working on your functions, changes will be reflected instantly/automatically. If you used sam build before running local commands, you will need to re-run sam build for the changes to be picked up. You only need to restart SAM CLI if you update your AWS SAM template
2023-04-14 12:33:01,475 | Localhost server is starting up. Multi-threading = True
 * Tip: There are .env or .flaskenv files present. Do "pip install python-dotenv" to use them.
2023-04-14 12:33:01 WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:3001
2023-04-14 12:33:01 Press CTRL+C to quit
```

Whereas this is the output when I trigger any lambda via postman:
```
2023-04-14 12:46:58,924 | Constructed String representation of Event to invoke Lambda. Event: {"body": null, "headers": {"Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7", "Accept-Encoding": "gzip, deflate, br", "Accept-Language": "en-US,en;q=0.9,it-IT;q=0.8,it;q=0.7,ru;q=0.6", "Connection": "keep-alive", "Cookie": "username-127-0-0-1-8888=\"2|1:0|10:1678953151|23:username-127-0-0-1-8888|44:OWI4NTJmYzYzNGY5NDVmYWJiODU5MWI4ZTg0NDI3MjE=|e541be0153eaa06a1d0c74ba46a0538b5b5b49d93aff804ad634ad8f355fc1e2\"", "Dnt": "1", "Host": "127.0.0.1:3001", "Sec-Ch-Ua": "\"Chromium\";v=\"112\", \"Google Chrome\";v=\"112\", \"Not:A-Brand\";v=\"99\"", "Sec-Ch-Ua-Mobile": "?0", "Sec-Ch-Ua-Platform": "\"Windows\"", "Sec-Fetch-Dest": "document", "Sec-Fetch-Mode": "navigate", "Sec-Fetch-Site": "none", "Sec-Fetch-User": "?1", "Upgrade-Insecure-Requests": "1", "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36", "X-Forwarded-Port": "3001", "X-Forwarded-Proto": "http"}, "httpMethod": "GET", "isBase64Encoded": false, "multiValueHeaders": {"Accept": ["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"], "Accept-Encoding": ["gzip, deflate, br"], "Accept-Language": ["en-US,en;q=0.9,it-IT;q=0.8,it;q=0.7,ru;q=0.6"], "Connection": ["keep-alive"], "Cookie": ["username-127-0-0-1-8888=\"2|1:0|10:1678953151|23:username-127-0-0-1-8888|44:OWI4NTJmYzYzNGY5NDVmYWJiODU5MWI4ZTg0NDI3MjE=|e541be0153eaa06a1d0c74ba46a0538b5b5b49d93aff804ad634ad8f355fc1e2\""], "Dnt": ["1"], "Host": ["127.0.0.1:3001"], "Sec-Ch-Ua": ["\"Chromium\";v=\"112\", \"Google Chrome\";v=\"112\", \"Not:A-Brand\";v=\"99\""], "Sec-Ch-Ua-Mobile": ["?0"], "Sec-Ch-Ua-Platform": ["\"Windows\""], "Sec-Fetch-Dest": ["document"], "Sec-Fetch-Mode": ["navigate"], "Sec-Fetch-Site": ["none"], "Sec-Fetch-User": ["?1"], "Upgrade-Insecure-Requests": ["1"], "User-Agent": ["Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36"], "X-Forwarded-Port": ["3001"], "X-Forwarded-Proto": ["http"]}, "multiValueQueryStringParameters": null, "path": "/info", "pathParameters": null, "queryStringParameters": null, "requestContext": {"accountId": "123456789012", "apiId": "1234567890", "domainName": "127.0.0.1:3001", "extendedRequestId": null, "httpMethod": "GET", "identity": {"accountId": null, "apiKey": null, "caller": null, "cognitoAuthenticationProvider": null, "cognitoAuthenticationType": null, "cognitoIdentityPoolId": null, "sourceIp": "127.0.0.1", "user": null, "userAgent": "Custom User Agent String", "userArn": null}, "path": "/info", "protocol": "HTTP/1.1", "requestId": "fa217757-13dd-430e-9da1-6d21d5c95c74", "requestTime": "14/Apr/2023:10:46:53 +0000", "requestTimeEpoch": 1681469213, "resourceId": "123456", "resourcePath": "/info", "stage": "dev"}, "resource": "/info", "stageVariables": null, "version": "1.0"}
2023-04-14 12:46:58,926 | Found one Lambda function with name 'PrintRequestContext'
2023-04-14 12:46:58,927 | Invoking ./dist/functions/api-gateway/print-request-context/handler.handler (nodejs18.x)
2023-04-14 12:46:58,927 | No environment variables found for function 'PrintRequestContext'
2023-04-14 12:46:58,928 | Loading AWS credentials from session with profile 'None'
2023-04-14 12:46:59,032 | Resolving code path. Cwd=/home/andrea/authentication-template/backend, CodeUri=/home/andrea/authentication-template/backend
2023-04-14 12:46:59,033 | Resolved absolute path to code is /home/andrea/authentication-template/backend
2023-04-14 12:46:59,034 | Code /home/andrea/authentication-template/backend is not a zip/jar file
2023-04-14 12:47:02,095 | Local image is up-to-date
2023-04-14 12:47:02,118 | Using local image: public.ecr.aws/lambda/nodejs:18-rapid-x86_64.

2023-04-14 12:47:02,118 | Mounting /home/andrea/authentication-template/backend as /var/task:ro,delegated, inside runtime container
2023-04-14 12:47:02,383 | Cleaning all decompressed code dirs
2023-04-14 12:47:02,384 | Exception on /info [GET]
Traceback (most recent call last):
  File "docker/api/client.py", line 261, in _raise_for_status
  File "requests/models.py", line 1021, in raise_for_status
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http+docker://localhost/v1.35/containers/cf7cead11105f34bc4044f6536fcaf0a23ccb19b5d40627e3ab7babb5a47f5e3/start

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "flask/app.py", line 2073, in wsgi_app
  File "flask/app.py", line 1518, in full_dispatch_request
  File "flask/app.py", line 1516, in full_dispatch_request
  File "flask/app.py", line 1502, in dispatch_request
  File "samcli/local/apigw/local_apigw_service.py", line 364, in _request_handler
  File "samcli/commands/local/lib/local_lambda.py", line 151, in invoke
  File "samcli/lib/telemetry/metric.py", line 323, in wrapped_func
  File "samcli/local/lambdafn/runtime.py", line 190, in invoke
  File "samcli/local/lambdafn/runtime.py", line 145, in run
  File "samcli/local/docker/manager.py", line 116, in run
  File "samcli/local/docker/container.py", line 316, in start
  File "docker/models/containers.py", line 400, in start
  File "docker/utils/decorators.py", line 19, in wrapped
  File "docker/api/container.py", line 1095, in start
  File "docker/api/client.py", line 263, in _raise_for_status
  File "docker/errors.py", line 31, in create_api_error_from_http_exception
docker.errors.APIError: 500 Server Error: Internal Server Error ("b'Ports are not available: exposing port TCP 127.0.0.1:6921 -> 0.0.0.0:0: listen tcp 127.0.0.1:6921: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.'")
2023-04-14 12:47:02 127.0.0.1 - - [14/Apr/2023 12:47:02] "GET /info HTTP/1.1" 502 -

```

### Additional environment details
```
{
  "version": "1.78.0",
  "system": {
    "python": "3.7.10",
    "os": "Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-debian-bullseye-sid"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.24",
    "aws_cdk": "Not available",
    "terraform": "Not available"
  }
}
```

Any hints?

## Addition
Running the command to invoke only the lambda locally
```
sam local invoke HelloWorld -e src/functions/api-gateway/hello-world/events/working-event.json --debug
```
returns the following

```
2023-04-14 13:03:11,083 | Config file location: /home/andrea/authentication-template/backend/samconfig.toml
2023-04-14 13:03:11,083 | Config file '/home/andrea/authentication-template/backend/samconfig.toml' does not exist
2023-04-14 13:03:11,124 | Using SAM Template at /home/andrea/authentication-template/backend/template.yml
2023-04-14 13:03:11,183 | Using config file: samconfig.toml, config environment: default
2023-04-14 13:03:11,183 | Expand command line arguments to:
2023-04-14 13:03:11,183 | --template_file=/home/andrea/authentication-template/backend/template.yml --event=src/functions/api-gateway/hello-world/events/working-event.json --function_logical_id=HelloWorld --no_event --layer_cache_basedir=/home/andrea/.aws-sam/layers-pkg --container_host=localhost --container_host_interface=127.0.0.1 
2023-04-14 13:03:11,183 | local invoke command is called
2023-04-14 13:03:11,216 | No Parameters detected in the template
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource AuthStackBucket, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource Api, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource HelloWorld, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource PrintRequestContext, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource KMSKey, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource UserPool, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource SESEmailIdentity, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,240 | There is no customer defined id or cdk path defined for resource UserPoolClient, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,241 | There is no customer defined id or cdk path defined for resource PreSignup, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,241 | There is no customer defined id or cdk path defined for resource CustomEmailSender, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,241 | There is no customer defined id or cdk path defined for resource OnCognitoSignupPermission, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,241 | There is no customer defined id or cdk path defined for resource AuthTemplateTable, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,241 | There is no customer defined id or cdk path defined for resource CreateOrganization, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,244 | 0 stacks found in the template
2023-04-14 13:03:11,245 | No Parameters detected in the template
2023-04-14 13:03:11,300 | There is no customer defined id or cdk path defined for resource AuthStackBucket, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,300 | There is no customer defined id or cdk path defined for resource Api, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,300 | There is no customer defined id or cdk path defined for resource HelloWorld, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,300 | There is no customer defined id or cdk path defined for resource PrintRequestContext, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource KMSKey, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource UserPool, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource SESEmailIdentity, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource UserPoolClient, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource PreSignup, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource CustomEmailSender, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource OnCognitoSignupPermission, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource AuthTemplateTable, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,301 | There is no customer defined id or cdk path defined for resource CreateOrganization, so we will use the resource logical id as the resource id
2023-04-14 13:03:11,303 | 13 resources found in the stack 
2023-04-14 13:03:11,304 | Found Serverless function with name='HelloWorld' and CodeUri='.'
2023-04-14 13:03:11,304 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 13:03:11,304 | Found Serverless function with name='PrintRequestContext' and CodeUri='.'
2023-04-14 13:03:11,304 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 13:03:11,304 | Found Serverless function with name='PreSignup' and CodeUri='.'
2023-04-14 13:03:11,304 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 13:03:11,304 | Found Serverless function with name='CustomEmailSender' and CodeUri='.'
2023-04-14 13:03:11,304 | --base-dir is not presented, adjusting uri . relative to /home/andreaauthentication-template/backend/template.yml
2023-04-14 13:03:11,304 | Found Serverless function with name='CreateOrganization' and CodeUri='.'
2023-04-14 13:03:11,304 | --base-dir is not presented, adjusting uri . relative to /home/andrea/authentication-template/backend/template.yml
2023-04-14 13:03:11,313 | Found one Lambda function with name 'HelloWorld'
2023-04-14 13:03:11,314 | Invoking ./dist/functions/api-gateway/hello-world/handler.handler (nodejs18.x)
2023-04-14 13:03:11,314 | No environment variables found for function 'HelloWorld'
2023-04-14 13:03:11,314 | Loading AWS credentials from session with profile 'None'
2023-04-14 13:03:11,341 | Resolving code path. Cwd=/home/andrea/authentication-template/backend, CodeUri=/home/andrea/authentication-template/backend
2023-04-14 13:03:11,341 | Resolved absolute path to code is /home/andrea/authentication-template/backend
2023-04-14 13:03:11,341 | Code /home/andrea/authentication-template/backend is not a zip/jar file
2023-04-14 13:03:14,011 | Local image is up-to-date
2023-04-14 13:03:14,026 | Using local image: public.ecr.aws/lambda/nodejs:18-rapid-x86_64.

2023-04-14 13:03:14,026 | Mounting /home/andrea/authentication-template/backend as /var/task:ro,delegated, inside runtime container
2023-04-14 13:03:14,355 | Cleaning all decompressed code dirs
2023-04-14 13:03:14,356 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2023-04-14 13:03:14,373 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': '547def4f-eb06-4bea-becf-75f73a290c7e', 'installationId': '91df9453-a2e7-4bf7-9131-b3ae279fcd27', 'sessionId': 'c3673043-66d9-4859-b744-b2bff1e76932', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.7.10', 'samcliVersion': '1.78.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam local invoke', 'metricSpecificAttributes': {'projectType': 'CFN', 'gitOrigin': None, 'projectName': '10e08a419e850eba1ebba18fdd28eb7ec1b7e8baa9bcc3b973e2b8891ec726be', 'initialCommit': None}, 'duration': 3173, 'exitReason': 'APIError', 'exitCode': 255}}]}
2023-04-14 13:03:15,072 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)

Error: 500 Server Error: Internal Server Error ("b'Ports are not available: exposing port TCP 127.0.0.1:7001 -> 0.0.0.0:0: listen tcp 127.0.0.1:7001: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.'")
Traceback:
  File "click/core.py", line 1055, in main
  File "click/core.py", line 1657, in invoke
  File "click/core.py", line 1657, in invoke
  File "click/core.py", line 1404, in invoke
  File "click/core.py", line 760, in invoke
  File "click/decorators.py", line 84, in new_func
  File "click/core.py", line 760, in invoke
  File "samcli/lib/telemetry/metric.py", line 183, in wrapped
  File "samcli/lib/telemetry/metric.py", line 148, in wrapped
  File "samcli/lib/utils/version_checker.py", line 42, in wrapped
  File "samcli/cli/main.py", line 92, in wrapper
  File "samcli/commands/local/invoke/cli.py", line 118, in cli
  File "samcli/commands/local/invoke/cli.py", line 203, in do_cli
  File "samcli/commands/local/lib/local_lambda.py", line 151, in invoke
  File "samcli/lib/telemetry/metric.py", line 323, in wrapped_func
  File "samcli/local/lambdafn/runtime.py", line 190, in invoke
  File "samcli/local/lambdafn/runtime.py", line 145, in run
  File "samcli/local/docker/manager.py", line 116, in run
  File "samcli/local/docker/container.py", line 316, in start
  File "docker/models/containers.py", line 400, in start
  File "docker/utils/decorators.py", line 19, in wrapped
  File "docker/api/container.py", line 1095, in start
  File "docker/api/client.py", line 263, in _raise_for_status
  File "docker/errors.py", line 31, in create_api_error_from_http_exception
```

**Comment by jfuss** [comment]
@andrybicio Looks like the port we selected is in use: `docker.errors.APIError: 500 Server Error: Internal Server Error ("b'Ports are not available: exposing port TCP 127.0.0.1:6921 -> 0.0.0.0:0: listen tcp 127.0.0.1:6921: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.'")`

Does this happen all the time or just "randomly"?

**Comment by jfuss** [comment]
This is a duplicate of https://github.com/aws/aws-sam-cli/issues/4952. We still don't know what is exactly causing this but looks like it has something to do with WSL2. One thing to try is downgrading to a previous version of Docker. We saw some customers have unexpected failures in the latest version. While that doesn't directly solve this issue, may give you a way to make progress while we dive deeper.

Closing as a duplicate.
